backend/src/transaction_event/protocol_level_tokens.rs

```diff
@@ -644,20 +649,73 @@ impl PreparedTokenUpdate {
         match self.event_type {
             TokenUpdateEventType::Mint => {
                 if let Some(ref target) = self.target {
+                    let account_address =
+                        concordium_rust_sdk::base::contracts_common::AccountAddress::from_str(
+                            target.as_str(),
+                        )
+                        .map_err(|_| {
+                            anyhow::anyhow!(
+                                "Failed to convert account address to canonical address: {}",
+                                target
+                            )
+                        })?;
```

Comment:
It would be better to do it as part of the preprocessing and not when this is stored to the database.
Also you should not need to convert it into a string and then parse the string again.

Reply:
We will get back here, I was trying to do that , but I was getting stuck due to some errors and seems like a lot of change , I will make a card for this in the backlog.